### PR TITLE
restya-board: further fix for functionality

### PR DESCRIPTION
**Pull request by nek0**
###### Motivation for this change
I introduced an error in configuration in #74763 which is fixed in this PR.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @aanderse @c0bw3b

### CODE REVIEW

File: nixos/modules/services/web-apps/restya-board.nix [hunk 1]
```diff
@@ -285,7 +285,7 @@ in
           sed -i "s/^.*'R_DB_PASSWORD'.*$/define('R_DB_PASSWORD', 'restya');/g" "${runDir}/server/php/config.inc.php"
         '' else ''
           sed -i "s/^.*'R_DB_HOST'.*$/define('R_DB_HOST', '${cfg.database.host}');/g" "${runDir}/server/php/config.inc.php"
-          sed -i "s/^.*'R_DB_PASSWORD'.*$/define('R_DB_PASSWORD', ${if cfg.database.passwordFile == null then "''" else "'file_get_contents(${cfg.database.passwordFile})'"});/g" "${runDir}/server/php/config.inc.php
+          sed -i "s/^.*'R_DB_PASSWORD'.*$/define('R_DB_PASSWORD', ${if cfg.database.passwordFile == null then "''" else "'${builtins.replaceStrings ["\n"] [""] (builtins.readFile cfg.database.passwordFile)}');/g"}" "${runDir}/server/php/config.inc.php"
```
Comment (aanderse) [inline nixos/modules/services/web-apps/restya-board.nix]: `builtins.readFile` would read the file into the nix store, making the `passwordFile` viewable by all users on the machine.

File: nixos/modules/services/web-apps/restya-board.nix [hunk 2]
```diff
@@ -305,9 +305,9 @@ in
         mkdir -m 0750 -p "${cfg.dataDir}/client/img"
         cp -r "${pkgs.restya-board}/media/"* "${cfg.dataDir}/media"
         cp -r "${pkgs.restya-board}/client/img/"* "${cfg.dataDir}/client/img"
-        chown "${cfg.user}"."${cfg.group}" "${cfg.dataDir}"
-        chown -R "${cfg.user}"."${cfg.group}" "${cfg.dataDir}/media"
-        chown -R "${cfg.user}"."${cfg.group}" "${cfg.dataDir}/client/img"
+        chown "${cfg.user}":"${cfg.group}" "${cfg.dataDir}"
```
Comment (aanderse) [inline nixos/modules/services/web-apps/restya-board.nix]: What is the purpose of `.` -> `:`?

**Review by aanderse — CHANGES_REQUESTED** [review]

File: nixos/modules/services/web-apps/restya-board.nix (hunk 2 above)
Comment (nek0) [reply nixos/modules/services/web-apps/restya-board.nix]: I don't really know where `.` comes from, but `:` is the only correct syntax as stated by the manpage.

File: nixos/modules/services/web-apps/restya-board.nix (hunk 1 above)
Comment (nek0) [reply nixos/modules/services/web-apps/restya-board.nix]: Then there is a problem. Both the php function doing it beforehand and `builtins.readFile` both appended a newline character at the end of the password, which is not present in the file.

File: nixos/modules/services/web-apps/restya-board.nix (hunk 2 above)
Comment (aanderse) [reply nixos/modules/services/web-apps/restya-board.nix]: Fair enough. `:` is legacy, so OK.

**Comment by nek0** [comment]
Ok. Using simple `cat` does not yieald any strange newline characters.
Additionally I found this hard dependency on a local postgresql database, which I loosened up.

File: nixos/modules/services/web-apps/restya-board.nix (hunk 1 above)
Comment (nek0) [reply nixos/modules/services/web-apps/restya-board.nix]: ok. a simple `cat` does the trick.

**Comment by Artturin** [comment]
@ofborg eval

**Comment by Artturin** [comment]
use squash and merge when merging this 
image

**Comment by infinisil** [comment]
This looks fine, even without a squash (what was I thinking?). Sorry it took so long, thanks for the fix!

**Pull request merged**